pkg/environment/env.go

```diff
@@ -41,8 +41,18 @@ const (
 
 // Mirror return mirror of tiup.
 // If it's not defined, it will use "https://tiup-mirrors.pingcap.com/".
-func Mirror() string {
-	if m := os.Getenv(repository.EnvMirrors); m != "" {
+func Mirror(cfg *localdata.TiUPConfig) string {
+	m := os.Getenv(repository.EnvMirrors)
+	if cfg != nil && cfg.Mirror != "" {
+		if m != "" && m != cfg.Mirror {
+			fmt.Printf(`WARNING: both mirror config and TIUP_MIRRORS have been set.
```

Comment:
It's better to call `ResetMirror` to switch the mirror to the latest environment variable and output some message to make the user realize that the mirror has been switched.